### ISSUE TITLE
Unswitch causes divergent syncthreads, which is not supported on pre-Volta architectures.

### ISSUE THREAD

**Issue by naoyam** [csarofeen/pytorch#910]
On P100, `BNRepro`, `BNRepro2`, `MagicSchedulerBatchnormalization` fail at validation. These Python tests also fail: `test_native_layer_norm`, `test_native_layer_norm_half`, `test_batch_norm` and `test_batch_norm_half`. Memcheck doesn't detect anything, but racecheck does. 

It's highly likely this is caused by the divergent syncthreads (actually `__barrier_sync`), which isn't supported on pre-Volta architectures. It is again unfortunate that syncthreads doesn't make the tests hang, but with `PYTORCH_NVFUSER_USE_BLOCK_SYNC_ATOMIC=1`, they do hang. 

The decision is to disable the fuser on pre-Volta devices as our focus is on future architectures. Technically, it's possible to tweak the scheduling to avoid these fusions only on pre-Volta devices, but that doesn't align with our priorities.

**Comment by jjsjann123** [comment]
Sounds like it's only normalization that's causing trouble. But softmax doesn't seem to show up here. 😕 

Do we think reduction would be safe?

**Comment by naoyam** [comment]
Anything making `syncthreads` inside predicates using `threadIdx` is not safe on pre-Volta, including `unswitch` and `unroll` with reduction and broadcast. It may work fine depending on values of `blockDim`.

I thought we are completely disabling nvfuser on pre-Volta, not just particular tests. Pinging @csarofeen for confirmation.

**Comment by jjsjann123** [comment]
Kindly pinging @csarofeen again on this.

I temporarily disabled normalization fusion on our container, but we would want a proper fix in our container.

**Comment by naoyam** [comment]
This is no longer a concern as we don't support pre-Volta devices.
